index.html

```diff
@@ -1832,12 +1756,17 @@ <h3>Key types and formats</h3>
 }
         </pre>
 
-        <p>
-If a public key does not exist in the <a>DID document</a>, the key was revoked
-or is invalid. The <a>DID document</a> MUST NOT express revoked keys using a
-<a>verification relationship</a>. Each <a>DID method</a> specification is
-expected to detail how revocation is performed and tracked.
-        </p>
+
+<p>The <a>DID document</a> MUST NOT express revoked keys using a
+<a>verification relationship</a>.
+</p> 
+<p>
+If a verification method does not exist in the <a>DID document</a>, the method was revoked
+or is invalid. 
```

Comment:
Another option I would be ok with that I think says less but doesn't create the problems I mentioned would be this:

```suggestion
Proofs provided by the <a>DID subject</a> must be verifiable by a verification method
that is in the <a>DID Document</a>. See <a href="#did-controller"></a> for
more information on verification methods for proofs that are not provided by the
<a>DID subject</a>.
```

Reply:
@OR13, please take a look at the above new suggestion. I think I found a way to avoid the contradictions without introducing new language that isn't already in the spec.

Reply:
I am stuck on the words "provided by"....

I think bringing did subject and did controller in this way is harmful.

I suggest first stating:

1. "If the verification method for verifying a proof is not in the <a>DID Document</a> that results from resolving that verification method, then that verification method is considered invalid or revoked for that proof."

Then adding something to the effect of 

2. "verification methods for a subject can be controlled by the subject or a third party controller. See
<a href="#did-controller"></a> for more information on verification methods
for proofs that are not provided by the <a>DID subject</a>."

1 applies to both subject === controller and subject !== controller.

2 is essentially a warning that controller might not be did subject... which is a case for which we have 0 examples, and which is clearly not well understood (KERI is reinventing this concept IMO)... I am in favor of providing much better clarity on 2, including some actual examples.... I would suggest we raise an issue for it / mark it at risk as well.

Reply:
@dlongley I would accept your proposed changes, as is if you can raise an issue for this, noting we are still discussing it.

Reply:
@OR13,

> I am stuck on the words "provided by"...

That is hard fought language to reach consensus from elsewhere in the spec (emphasis mine):

> When a controller property is present in a DID Document, its value expresses one or more DIDs. Any verification methods contained in the DID Documents for those DIDs SHOULD be accepted as authoritative, such that proofs that satisfy those verification methods are to be considered equivalent to proofs **provided by the DID Subject**.

This is why I used it -- to ensure it's consistent. That being said, your suggestion may help us avoid all of this. I don't think we even need to talk about proofs now but I've tweaked your suggestion to make it more correct, IMO. This is about *dereferencing* verification methods there were referenced elsewhere (in proofs, for example) and then not finding them int the DID Document where they should be:

```suggestion
If a referenced verification method is not in the <a>DID Document</a> used
dereference it, then that verification method is considered invalid or revoked.
```

What do you think?

Reply:
I am good with that language, but I think you are missing a "to"...

> If a referenced verification method is not in the <a>DID Document</a> used to
dereference it, then that verification method is considered invalid or revoked.

Reply:
@OR13, added the missing "to", thanks!